var/spack/repos/builtin/packages/flux-pmix/package.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack import *
+
+
+class FluxPmix(AutotoolsPackage):
+    """ A flux shell plugin to bootstrap openmpi v5+ """
+
+    homepage = "https://github.com/flux-framework/flux-pmix"
+    url      = "https://github.com/flux-framework/flux-pmix/releases/download/v0.1.0/flux-pmix-0.1.0.tar.gz"
+    git      = "https://github.com/flux-framework/flux-pmix.git"
+
+    maintainers = ['grondo']
+
+    version('main', branch='main')
+    version('0.2.0', sha256='d09f1fe6ffe54f83be4677e1e727640521d8110090515d94013eba0f58216934')
+
+    depends_on('flux-core@0.30.0:')
+    depends_on('pmix@v4.1.0:')
+    depends_on('openmpi')
+
+    #  Some older pkg-config versions cause errors with ompi.pc file,
+    #   so require latest version from spack:
+    depends_on('pkg-config', type='build')
```

Comment:
In general it is best to state the real requirements to build a software in `package.py`, regardless of what you might find on most systems. Then if there's a need to cut build-time you can:
```console
spack external find <pkg>
``` 
to use what is installed already outside of Spack. Are there other build dependencies that are needed and not listed?

Reply:
I don't think there are any extra runtime dependencies over what is listed in the `package.py` 

```console
 grondo@quartz1538:~/git/spack$ spack external find flux-pmix
==> No new external packages detected
```
Thank you for the hint about the useful command.

Reply:
Actually, looking again, the package does explicitly require `libczmq` and `jansson`, but these are transitively  satisfied by the `flux-core` dependency. Is this ok, or do you prefer these be added to the `package.py`?

Reply:
What I meant is that `pkg-config` and maybe `auto*` packages should be listed as build dependencies if they are needed. Then you can:
```console
% spack external find pkg-config
```
if you want to reuse `pkg-config` from your host instead of having Spack build it. I was pointing this out since the impression I got from the comment was that `pkg-config` was added only because some system ship a version that it too old, while my advice is to always add it if it is a dependency.

Reply:
Ah, I understand now. That makes sense, thanks. 

I will remove the confusing comment then. I have already added the autotools as a build dependency.

Apologies, I'm a spack newbie!

Reply:
No worries, thanks @grondo !